File: DecompiledGameScripts/cyberpunk/network/systems/surveillanceSystem.swift

public class SurveillanceSystem extends DeviceSystemBase {

  protected cb func OnRequestComponents(ri: EntityRequestComponentsInterface) -> Bool {
    super.OnRequestComponents(ri);
  }

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as SurveillanceSystemController;
    super.OnTakeControl(ri);
  }

  protected const func GetController() -> ref<SurveillanceSystemController> {
    return this.m_controller as SurveillanceSystemController;
  }

  public const func GetDevicePS() -> ref<SurveillanceSystemControllerPS> {
    return this.GetController().GetPS();
  }
}

File: Definitions/Game_Definitions/classes/SurveillanceSystem.lua
---@meta
---@diagnostic disable

---@class SurveillanceSystem : DeviceSystemBase
SurveillanceSystem = {}

---@return SurveillanceSystem
function SurveillanceSystem.new() return end

---@param props table
---@return SurveillanceSystem
function SurveillanceSystem.new(props) return end

---@param ri entEntityRequestComponentsInterface
---@return Bool
function SurveillanceSystem:OnRequestComponents(ri) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function SurveillanceSystem:OnTakeControl(ri) return end

---@return SurveillanceSystemController
function SurveillanceSystem:GetController() return end

---@return SurveillanceSystemControllerPS
function SurveillanceSystem:GetDevicePS() return end

